src/main/java/org/emela/chautari/service/impl/RentalResourceServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+    private ResourceEntityRepository resourceEntityRepository;
+
+    public RentalResourceServiceImpl(UserService userService, RentalService rentalService,
+                                     ResourceEntityRepository resourceEntityRepository) {
+        this.userService = userService;
+        this.rentalService = rentalService;
+        this.resourceEntityRepository = resourceEntityRepository;
+    }
+
+
+    @Override
+    public ResourceResponse uploadResource(String userId, String rentalId, MultipartFile file) {
+        UserEntity user = userService.getUserEntity(userId);
+        RentalEntity rentalEntity = rentalService.getRentalEntity(rentalId);
+
+        ResourceEntity resourceEntity = BuildResourceEntity(file, user, rentalEntity);
+
+        return buildResourceResponse(resourceEntityRepository.save(resourceEntity).getResourceId());
+    }
+
+    @Override
+    public ResourceResponseBean uploadResources(String userId, String rentalId, List<MultipartFile> files) {
+        UserEntity user = userService.getUserEntity(userId);
+        RentalEntity rentalEntity = rentalService.getRentalEntity(rentalId);
+
+        List<ResourceEntity> resourceEntities = files.stream()
+                .map(it -> BuildResourceEntity(it, user, rentalEntity)).collect(Collectors.toList());
+
+        List<ResourceEntity> savedEntities = resourceEntityRepository.saveAll(resourceEntities);
+
+        List<ResourceResponse> resourceResponses = CollectionUtils.emptyIfNull(savedEntities).stream()
+                .map(it -> buildResourceResponse(it.getResourceId())).collect(Collectors.toList());
+
+        return new ResourceResponseBean().message("success").resourceResponse(resourceResponses).status("success");
+    }
+
+    @Override
+    public ResourceDetail getResource(String resourceId, String rentalId) {
+        ResourceEntity resourceDetail;
+        try {
+            resourceDetail = resourceEntityRepository.findByResourceId(UUID.fromString(resourceId))
+                    .orElseThrow(ResourceNotFoundException::new);
+            return new ResourceDetail().resourceId(resourceDetail.getResourceId().toString())
+                    .file(resourceDetail.getResourceFile()).message("Success");
+        } catch (Exception e) {
+            log.error("resource not found", e);
+            throw new RentalServiceException(e);
+        }
+    }
+
+    @Override
+    public ResourceResponse deleteResource(String resourceId, String rentalId) {
+        log.info("resource delete request has received for resource-id {} ", resourceId);
+        try {
+            ResourceEntity resourceEntity = resourceEntityRepository.findByResourceId(UUID.fromString(resourceId))
+                    .orElseThrow(ResourceNotFoundException::new);
+            resourceEntityRepository.delete(resourceEntity);
+        } catch (Exception e) {
+            log.error("Delete resource Item failed", e);
+            throw new RentalServiceException(e);
```

Comment:
Can you pass the custom exception with some known unique identifier (resourceId, userId, rental-Id) message when you throwing the exception?

So that it would be easier to track down a specific transaction.

```java
    throw new RentalServiceException("Delete resource Item failed for rental id " + rentalId + " resourceId  " + resourceId  , e);
```